src/main/java/com/smattme/MysqlExportService.java

```diff
@@ -338,14 +347,32 @@ private String exportToSql() throws SQLException {
         TablesResponse allTablesAndViews = MysqlBaseService.getAllTablesAndViews(database, stmt);
 
         List<String> tables = allTablesAndViews.getTables();
-        //for every table, get the table creation and data
-        // insert statement
-        for (String s: tables) {
-            try {
-                sql.append(getTableInsertStatement(s.trim()));
-                sql.append(getDataInsertStatement(s.trim()));
-            } catch (SQLException e) {
-                logger.error("Exception occurred while processing table: " + s, e);
+
+        // Retrieve specific tables from properties, if they exist
+        String specificTables = properties.containsKey(SPECIFIC_TABLES_FOR_EXPORT)
+            ? properties.getProperty(SPECIFIC_TABLES_FOR_EXPORT)
+            : "";
+
+        // Split the specific tables into a list
+        List<String> specificTableList = new ArrayList<>();
+        if (!specificTables.isEmpty()) {
+            specificTableList = Arrays.stream(specificTables.split(",")).map(String::trim).toList();
+        }
+
+        for (String s : tables) {
+
+            // Trim whitespace from the table name
+            String specificTableName = s.trim();
+
+            // Determine if the specific table list is empty or includes the current table name
+            // If so, generate SQL statements for table creation and data insertion
+            if (specificTableList.isEmpty() || specificTableList.contains(specificTableName)) {

```

Comment:
This logic is better placed in the function `com.smattme.MysqlBaseService#getAllTablesAndViews`. Since the function is the one responsible for determining which tables will be added to list of `tables` for processing or not.

```
ResultSet rs;
        rs = stmt.executeQuery("SHOW TABLE STATUS FROM `" + database + "`;");
        while ( rs.next() ) {
           String name = rs.getString("Name");
           //TODO check if the table/view name is in the specified list OR the list is empty before adding them to the final list
            String comment = rs.getString("Comment");
            if("VIEW".equals(comment)) {
                views.add(name);
            }
            else {
                tables.add(name);
            }
        }
```

This means you'll need to modify the signature of the com.smattme.MysqlBaseService#getAllTablesAndViews to expect a `Properties` object. Then, in it, you'll construct a table whitelist and use it in the while loop for building the table.

**It would be even more awesome if it is possible to add the whitelist to the `SHOW TABLE STATUS FROM database` query directly**.

Looking forward to your update